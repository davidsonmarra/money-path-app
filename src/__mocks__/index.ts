export const mockGoogleSignIn = {
    configure: jest.fn(),
    hasPlayServices: jest.fn(),
    signIn: jest.fn(),
};