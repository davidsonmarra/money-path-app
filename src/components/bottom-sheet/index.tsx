import React from 'react';
import { View, DimensionValue } from 'react-native';
import Modal from 'react-native-modal';
import Text, { TextType } from 'src/components/text';
import useStyles from 'src/components/bottom-sheet/styles';

export interface Props {
  isVisible: boolean;
  onClose: () => void;
  title: string;
  children: React.ReactNode;
  maxHeight?: DimensionValue;
  testID?: string;
}

const BottomSheet = ({
  isVisible,
  onClose,
  title,
  children,
  maxHeight = '80%',
  testID,
}: Props) => {
  const styles = useStyles({ maxHeight });

  return (
    <Modal
      isVisible={isVisible}
      onBackdropPress={onClose}
      onBackButtonPress={onClose}
      style={styles.modal}
      animationIn="slideInUp"
      animationOut="slideOutDown"
      animationInTiming={300}
      animationOutTiming={300}
      backdropTransitionInTiming={300}
      backdropTransitionOutTiming={300}
      testID={testID}
    >
      <View style={styles.bottomSheet}>
        <View style={styles.header}>
          <Text type={TextType.headingMedium} style={styles.headerTitle}>
            {title}
          </Text>
        </View>
        <View style={styles.content}>{children}</View>
      </View>
    </Modal>
  );
};

export default BottomSheet;
