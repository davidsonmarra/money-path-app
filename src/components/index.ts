export { default as Button, ButtonType } from 'src/components/button';
export {
  default as Divider,
  DividerSize,
  DividerType,
} from 'src/components/divider';
export { default as Feedback, FeedbackType } from 'src/components/feedback';
export { default as Header } from 'src/components/header';
export { default as Input } from 'src/components/input';
export { default as TextArea } from 'src/components/text-area';
export { default as ListItem } from 'src/components/list-item';
export { default as ProfileHeader } from 'src/components/profile-header';
export { default as Spacer } from 'src/components/spacer';
export { default as Text, TextType } from 'src/components/text';
export { default as IconRound } from 'src/components/icon-round';
export { default as StorybookView } from 'src/components/storybook-view';
export { default as SelectionGroup } from 'src/components/selection-group';
