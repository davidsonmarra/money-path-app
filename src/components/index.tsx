export {default as Button, ButtonType} from 'src/components/button';
export {default as Text, TextType} from 'src/components/text';
export {default as StorybookView} from 'src/components/storybook-view';
