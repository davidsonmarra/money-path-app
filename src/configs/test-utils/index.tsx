import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
