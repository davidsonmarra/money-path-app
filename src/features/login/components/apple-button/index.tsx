import React from 'react';
import { TouchableOpacityProps } from 'react-native';
import Button from 'src/components/button';
import { AppleIcon } from 'src/assets/icons';
import getStyles from 'src/features/login/components/apple-button/styles';

type Props = TouchableOpacityProps;

const AppleButton = ({ onPress }: Props) => {
  return (
    <Button
      onPress={onPress}
      leftIcon={<AppleIcon size={28} />}
      text="Entrar com Apple"
      style={getStyles().container}
    />
  );
};

export default AppleButton;
