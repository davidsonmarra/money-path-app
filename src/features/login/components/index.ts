export {default as AppleButton} from 'src/features/login/components/apple-button';
export {default as GoogleButton} from 'src/features/login/components/google-button';
