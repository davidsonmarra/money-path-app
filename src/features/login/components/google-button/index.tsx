import React from 'react';
import {TouchableOpacityProps} from 'react-native';
import {Button, ButtonType} from 'src/components';
import {GoogleIcon} from 'src/assets/icons';
import getStyles from 'src/features/login/components/google-button/styles';

type Props = TouchableOpacityProps;

const GoogleButton = ({onPress}: Props) => {
  return (
    <Button
      onPress={onPress}
      leftIcon={<GoogleIcon size={28} />}
      text="Entrar com Google"
      style={getStyles().container}
      type={ButtonType.secondary}
    />
  );
};

export default GoogleButton;
