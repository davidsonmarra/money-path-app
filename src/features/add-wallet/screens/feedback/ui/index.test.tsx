import React from 'react';
import {ActivityIndicator} from 'react-native';
import {Feedback, FeedbackType} from 'src/components';
import {fireEvent, render} from 'src/configs/test-utils';
import FeedbackContainer, {
  States,
} from 'src/features/add-wallet/screens/feedback/ui';

let mockState = States.loading;
const mockOnClose = jest.fn();
const mockOnPress = jest.fn();
const containerInstance = ({state = mockState}) =>
  render(
    <FeedbackContainer
      state={state}
      onClose={mockOnClose}
      onPress={mockOnPress}
    />,
  );

describe('FeedbackContainer', () => {
  beforeEach(() => {
    mockState = States.loading;
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({state: States.default}).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call onBackPress', () => {
    const container = containerInstance({});
    container.getByTestId('btn-left-icon').props.onClick();
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  describe('when state is default', () => {
    it('should call onPress', () => {
      const container = containerInstance({state: States.default});
      const button = container.getByText('Voltar para a tela inicial');
      fireEvent.press(button);
      expect(mockOnPress).toHaveBeenCalledTimes(1);
    });

    it('should render success state correctly', () => {
      const container = containerInstance({state: States.default}).root;
      const feedback = container.findAllByType(Feedback);
      expect(feedback[0].props.type).toBe(FeedbackType.success);
    });
  });

  describe('when state is error', () => {
    it('should call onPress', () => {
      const container = containerInstance({state: States.error});
      const button = container.getByText('Tentar novamente');
      fireEvent.press(button);
      expect(mockOnPress).toHaveBeenCalledTimes(1);
    });

    it('should render error state correctly', () => {
      const container = containerInstance({state: States.error}).root;
      const feedback = container.findAllByType(Feedback);
      expect(feedback[0].props.type).toBe(FeedbackType.error);
    });
  });

  describe('when state is loading', () => {
    it('should render loading state correctly', () => {
      const container = containerInstance({state: States.loading}).root;
      const feedback = container.findAllByType(ActivityIndicator);
      expect(feedback[0]).toBeTruthy();
    });
  });
});
