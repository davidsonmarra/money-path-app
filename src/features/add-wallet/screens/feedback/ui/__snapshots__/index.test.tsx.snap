// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackContainer should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-close"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    />
    <View />
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#232E38",
        "flex": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View />
      <View
        style={
          {
            "alignItems": "center",
            "gap": 16,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 90,
                "height": 180,
                "justifyContent": "center",
                "marginBottom": 16,
                "width": 180,
              },
              {
                "backgroundColor": "#A6CD4E30",
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={100}
            bbWidth={100}
            fill="#A6CD4E"
            focusable={false}
            height={100}
            meetOrSlice={0}
            minX={0}
            minY={-960}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 100,
                  "width": 100,
                },
              ]
            }
            testID="icon-check"
            vbHeight={960}
            vbWidth={960}
            width={100}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4289121614,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="m424-296 282-282-56-56-226 226-114-114-56 56zm56 216q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 24,
                "fontWeight": 600,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Carteira criada!
        </Text>
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontWeight": 400,
                },
              ],
              [
                {
                  "textAlign": "center",
                },
                {
                  "color": "#FFFFFFAA",
                },
              ],
            ]
          }
        >
          Sua carteira foi criada com sucesso. Agora, você pode adicionar transações.
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EAC43D",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
      >
        <View />
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "fontWeight": 600,
                },
              ],
              {
                "color": "#FFF",
              },
            ]
          }
        >
          Voltar para a tela inicial
        </Text>
        <View />
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
